File: Matura/pos-survival-kit-shareable/example-project/06_Janssen_animalShelterServer/src/main/java/janssen/spring/entities/Sales.java
package janssen.spring.entities;

import janssen.spring.JsonViews;
import com.fasterxml.jackson.annotation.JsonView;
import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.*;
import java.time.LocalDate;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Sales {
  
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @JsonView(JsonViews.Summary.class)
  protected long id;
  
  @NotNull
  @ManyToOne
  @JsonView(JsonViews.Summary.class)
  protected AnimalShelter shelter;
  
  @NotNull
  @ManyToOne
  @JsonView(JsonViews.Summary.class)
  protected Animal animal;
  
  @Positive
  @Column(nullable = false)
  @JsonView(JsonViews.Summary.class)
  protected double price;
  
  @NotNull
  @PastOrPresent
  @Column(nullable = false)
  @JsonView(JsonViews.Summary.class)
  protected LocalDate date;
  
}

File: Matura/pos-survival-kit-shareable/example-project/06_Janssen_animalShelterApp/app/src/main/java/janssen/android/models/Sales.kt
package janssen.android.models

import java.time.LocalDate

class Sales {
    
    var id: Long = 0
    var shelter: AnimalShelter? = null
    var animal: Animal? = null
    var price: Double? = 0.0
    var date: LocalDate? = null
    
    override fun toString(): String {
        return "Sales(id=$id)"
    }
    
}
